ambari/src/main/java/io/brooklyn/ambari/server/AmbariServerImpl.java

```diff
@@ -367,6 +379,178 @@ public Map apply(@Nullable String fqdn) {
+        Map<String, Object> result = MutableMap.of();
+
+        result.putAll(defaultArgs);
+        for (Map.Entry config: ambariAlertNotifications.entrySet()) {
+            if (config.getValue() instanceof Map && CollectionUtils.exists(((Map) config.getValue()).values(),NotNullPredicate.INSTANCE)) {
+                Map<String, Object> subresult =  MutableMap.of();
+                if (result.containsKey(config.getKey())) {
+                    subresult.putAll((Map<String, Object>) result.get(config.getKey()));
+                }
+                for (Object subconfig: ((Map) config.getValue()).entrySet()) {
+                    subresult.put((String) ((Map.Entry) subconfig).getKey(), ((Map.Entry) subconfig).getValue());
+                }
+                result.put((String) config.getKey(), subresult);
+            } else {
+                Object value = config.getValue() instanceof Map ? ((Map) config.getValue()).keySet().toArray()[0] : config.getValue();
+                result.put((String) config.getKey(), value);
+            }
+        }
+
+        return result;
+    }
+
+    public Map<String, Integer> listAlertTargets() {
+        AlertTargets alertTargets = restAdapter.create(AlertTargetEndpoint.class).listAlertNotifications();
+
+        Map<String, Integer> result = MutableMap.of();
+        for (Map<String, Object> item: alertTargets.getItems()) {
+            result.put((String) ((Map) item.get("AlertTarget")).get("name"), ((Double) ((Map) item.get("AlertTarget")).get("id")).intValue());
+        }
+
+        return result;
+    }
+
+    public void updateAlertTarget(Integer targetId, Map<String, Object> ambariAlertNotifications) {
+        ImmutableMap<String, Map<String, Object>> alertTargetRequest = ImmutableMap.of("AlertTarget", ambariAlertNotifications);
+
+        restAdapter.create(AlertTargetEndpoint.class).editAlertNotification(targetId, alertTargetRequest);
+    }
+
+    private Map<String, Object> getAmbariAlertsConfigFromProperties() {
+        ConfigBag args = new ConfigBag();
+        Map<String, Object> properties = MutableMap.of();
+        for (Map.Entry<ConfigKey<?>, Object> entry: getManagementContext().getConfig().getAllConfig().entrySet()) {
+            String key = entry.getKey().getName();
+            if (key.startsWith(AmbariCluster.AMBARI_ALERTS_CONFIG_PREFIX)) {
+                key = Strings.removeAllFromStart(key,AmbariCluster.AMBARI_ALERTS_CONFIG_PREFIX);
+
+                if (key.equals(AmbariCluster.AMBARI_ALERTS_NOTIFICATION_ALERT_STATES.getName())) {
+                    args.putStringKey(key,toArray((String) entry.getValue()));
+                } else if (key.equals(AmbariCluster.AMBARI_ALERTS_NOTIFICATION_RECIPIENTS.getName())){
+                    args.putStringKey(key,formatAmbariAlertsNotificationRecipients((String) entry.getValue()));
```

Comment:
Actually `formatAmbariAlertsNotificationRecipients` is needed because it workarounds the not unified query for alert notifications in Ambari.  
`ambari.dispatch.recipients` value should be a list as `alert_states` also should, but instead of expecting a list like for `alert_states`, for the recipients field only it expects a `String` `[element1, element2]`